conjure-lib/src/main/java/com/palantir/conjure/java/lib/ImmutableByteArray.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+/** An immutable {@code byte[]} wrapper. */
+public final class ImmutableByteArray {
```

Comment:
Maybe ImmutableBytes? “binary” translates a few different ways — sometimes as a stream — so would avoid the direct naming here. Also open to “ByteString” or just “Bytes”.

Reply:
`Bytes` is reasonable, though I prefer `Binary`. Other types have different java bindings based on context, `int`, `OptionalInt`, `List<Integer>` for example.
Bytes sounds a bit like a utility rather than a data object (see `com.google.common.primitives.Bytes`).

Reply:
why not just ByteArray? why don't prefix other classes with `Immutable` even if they are, e.g. ResourceIdentifier.